Generic table component based on react-window

Bruce-Hopkins opened the pull request:
Please DO NOT DELETE this template. Fill it. PR without your declaration that you licence your code under MIT will 
be discarded. 

Tips
----------------

 * Give the same name to the pull request as the name of ticket if possible
 * Make the title in form of: Pull request title (#<issue-id>)
 * Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 * Test your code throughly before PR.

Pull Request title
-------------------

     Pull request title (#)
#73 

Pull Request description
-------------------------

    Pull request description

Creates a table component using react-window-infinite-loader. You can pass the items to the table. The table size will be proportionate to the size of the array passed.


----

We need your declaration that your contribution is MIT licensed without it, we can not merge your code.
To make clear that you license your contribution under the MIT you have to acknowledge this by using the following check-box.

 - [ ] I hereby declare this contribution to be licenced under the [MIT license](https://opensource.org/licenses/MIT)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Tracardi/tracardi-gui?pullRequest=90) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Tracardi/tracardi-gui?pullRequest=90) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Tracardi/tracardi-gui?pullRequest=90) it.</sub>

atompie commented:
Hi @Bruce-Hopkins Could you provide an example how can I use this component, please.

I noticed one error.

```
Uncaught TypeError: items is undefined
    children table.js:58
    render index.esm.js:187
    React 4
    sentryWrapped helpers.ts:88
    React 13
    js index.js:25
    factory react refresh:6
    Webpack 3
```

This is due to the fact that the default item prop is not defined. Also, propTypes would help to find out the props types.

atompie commented on src/components/elements/table/table.js:
```diff
@@ -0,0 +1,70 @@
+export default function Table(props) {
+  const height = props.height | 200,
+  width = props.width | 300,
+  items = props.items,
```
I do not know what is the structure of the items prop. Therefore could not test it.

Bruce-Hopkins replied:
The items prop requires an array of strings. It will display it in a single column

atompie commented on src/components/elements/table/table.js:
```diff
@@ -0,0 +1,70 @@
+  }
+
+  return (
+    <Fragment>
```
Is there a way to customize the columns to be displayed?

Bruce-Hopkins replied:
What do you mean by that? As in the text of the first row?

atompie reviewed (commented):
Can I customize columns as described in the task?

```
<Table data={}>
  <Column header="Name">{name}</Column>
  <Column header="Surname">{surname}</Column>
</Table>
```

Bruce-Hopkins commented:
In order to create the table, you need to pass an array into it with the props property "items".

I'm not sure if there should be a default configuration to prevent errors. 

And sure, I can add prop types.

Bruce-Hopkins commented:
Sure I can change it to that.

atompie commented:
@Bruce-Hopkins Can you paste in comments an example of usage?

Bruce-Hopkins commented:
I changed the code to accommodate grids, although for some reason lazy loading isn't working. I removed it for the time being. I also added propTypes.
 
Usage example:
```js
  <Table rows={[1, 2, 3]} columns={[1, 2, 3]}/> 
```
